src/core/infer_trace.h

```diff
@@ -90,6 +98,10 @@ class InferenceTrace {
       const int64_t* shape, uint64_t dim_count,
       TRITONSERVER_MemoryType memory_type, int64_t memory_type_id)
   {
+    if (level_ < TRITONSERVER_TRACE_LEVEL_TENSORS) {
```

Comment:
Do not use '<'. level_ is a bitmask of enabled trace levels. So you need to mask TRACE_LEVEL_TENSORS and only report if that bit is set.